### DISCUSSION TITLE
Potential Refactoring

### DISCUSSION THREAD

**Discussion by alvintanjianjia** [dropwizard/dropwizard#5402] [Ideas]
Hi,

I am currently doing some research on Estimating Time Taken for Software Refactoring. Dropwizard is one of my case studies being a highly popular project.

I have identified some packages with refactoring potential and their estimated time to refactor. Would appreciate your kind feedback on the appropriateness of the estimated time to refactor.

1. io.dropwizard.migrations - 60 hours
2. io.dropwizard.logging - 58 hours
3. io.dropwizard.validation - 53 hours
4. io.dropwizard.util - 37 hours
5. io.dropwizard.hibernate - 32 hours

I will be grateful for any help you can provide.

Thank you.

Regards,
Alvin

**Comment by jplock** [comment]
What specifically are you estimating to be refactored?

**Comment by alvintanjianjia** [comment]
Hi,

I am specifically looking at potential to be refactored, such as breaking functions that are too long into shorter functions, or to remove code duplication. For the examples below, there are some duplicated code. 

**Example 1**
For io.dropwizard.migrations,
- NoAuthPrincipalEntityTest.java
- NoAuthPolymorphicPrincipalEntityTest.java

**Example 2**
For io.dropwizard.logging,
- LogbackAccessRequestLogFactory.java
- LogbackClassicRequestLogFactory.java

Would you agree that there could be some modification to the code to make it better? And if so, is the estimated time appropriate to carry out the improvements.

Thank you.

Regards,
Alvin

**Reply by rhowe** [reply]
If you look at my PR history for dropwizard you will see it is pretty much 100% refactoring (depending on your definition..). I've probably spent about 50-100 hours on it.

So if you run your algorithm on the before/after of those PRs you can check your maths!
